runbotics/runbotics-scheduler/src/websocket/websocket.service.ts

```diff
@@ -22,26 +22,44 @@ export class WebsocketService {
             throw new NotFoundException(`Bot ${botId} does not exist`);
         }
 
-        this.send(bot.installationId, message, body);
-    }
+        const connectedBots = this.botWebSocketGateway.connectedBots;
+        if (!connectedBots.length) {
+            this.logger.error('There are no connected bots');
+            throw new NotFoundException('There are no connected bots');
+        }
 
-    async sendMessageByInstallationId(
-        installationId: string, message: BotWsMessage, body?: any
-    ) {
-        const bot = await this.botService.findByInstallationId(installationId);
-        if (!bot) {
-            this.logger.error(`Bot ${installationId} does not exist`);
-            throw new NotFoundException(`Bot ${installationId} does not exist`);
+        const connectedBot = connectedBots.find((connectedBot) => connectedBot.botId === botId);
+        if (!connectedBot) {
+            this.logger.error(`There is no connection with bot ${botId}`);
+            throw new NotFoundException(`There is no connection with bot ${botId}`);
         }
 
-        this.send(bot.installationId, message, body);
+        try {
+            await this.send(connectedBot.socketId, message, body);
+        } catch (error) {
+            if (typeof error === 'string') {
+                throw new Error(error);
+            }
```

Comment:
what if error is not string?

Reply:
do we ignore error or is it thrown anyway? Is this condition even necessary? without try/catch error should be thrown anyway (?)

Reply:
Error requires parameter of type string, this is why I narrowed type here. I added type guard for response in acknowledgement callback, so now when promise is rejected, error message will always be of type string. Also, just in case, I added alternative throw with general message to give information that sending message to the bot failed.